types/reactstrap/lib/Collapse.d.ts

```diff
@@ -2,33 +2,38 @@ import * as React from 'react';
 import { CSSModule } from '../index';
 
 export interface CollapseProps extends React.HTMLProps<HTMLElement> {
-  [key: string]: any;
-  isOpen?: boolean;
-  classNames?: string;
-  cssModule?: CSSModule;
-  tag?: React.ReactType;
-  navbar?: boolean;
-  delay?: {
-    show: number
-    hide: number
-  };
-  onOpened?: () => void;
-  onClosed?: () => void;
-  onEntering?: () => void;
-  onEntered?: () => void;
-  onExit?: () => void;
-  onExiting?: () => void;
-  onExited?: () => void;
+    [key: string]: any;
+    isOpen?: boolean;
+    classNames?: string;
+    cssModule?: CSSModule;
+
+    tag?: string | React.ReactType;
+
+    navbar?: boolean;
+    delay?: {
+        show: number;
+        hide: number;
+    };
+    onOpened?: () => void;
+    onClosed?: () => void;
+    onEntering?: () => void;
+    onEntered?: () => void;
+    onExit?: () => void;
+    onExiting?: () => void;
+    onExited?: () => void;
 }
 
 export interface UncontrolledCollapseProps extends React.HTMLProps<HTMLElement> {
-  [key: string]: any;
-  isOpen?: boolean;
-  className?: string;
-  cssModule?: CSSModule;
-  tag?: React.ReactType;
-  navbar?: boolean;
-  toggle?: () => void;
+    [key: string]: any;
+    isOpen?: boolean;
+    className?: string;
+    cssModule?: CSSModule;
+
+    tag?: string | React.ReactType;
+
+    navbar?: boolean;
+
+    toggle?: React.KeyboardEventHandler<any> | React.MouseEventHandler<any>;
```

Comment:
Since we're here, I believe toggle shouldn't exist on `UncontrolledCollapse`.
I believe the full definition should be:

```typescript
export interface UncontrolledCollapseProps extends CollapseProps {
    defaultOpen?: PropTypes.bool;
    toggler: string;
    toggleEvents: Array<string>;
}
```

Might want to double check that although, I looked at it pretty quick.
https://github.com/reactstrap/reactstrap/blob/master/src/UncontrolledCollapse.js